sklearn/metrics/ranking.py

```diff
@@ -283,12 +283,22 @@ def detection_error_tradeoff_curve(y_true, y_score, pos_label=None,
     p_count = tps[-1]
     n_count = fps[-1]
 
-    # start with false positives is zero and stop with false negatives zero
-    # and reverse the outputs so list of false positives is decreasing
+    # start with false positives zero
+    first_ind = (
+        fps.searchsorted(fps[0], side='right') - 1
+        if fps.searchsorted(fps[0], side='right') > 0
+        else None
+    )
+    # stop with false negatives zero
     last_ind = tps.searchsorted(tps[-1]) + 1
-    first_ind = fps[::-1].searchsorted(fps[0])
-    sl = range(first_ind, last_ind)[::-1]
-    return fps[sl] / n_count, fns[sl] / p_count, thresholds[sl]
+    sl = slice(first_ind, last_ind)
```

Comment:
You can set the step so that the slice is reversed

Reply:
I initially went that route. Unfortunately that makes the computation of the slice borders very opaque. To deal with the fact that the "upper" slice border is exclusive we would have subtract `-2` in total with additional exception handling for the cases `1` and `0`.
I figured no one would ever be able follow what is happening and opted for the explicit approach.